new-packages/ts-project/src/state.ts

```diff
@@ -30,6 +29,40 @@ const createActionBlock = ({
     },
   };
 };
+export function createActorBlock({
+  sourceId,
+  parentId,
+  actorId,
+}: {
+  sourceId: string;
+  parentId: string;
+  actorId: string;
+}): ActorBlock {
+  const blockId = uniqueId();
+  return {
+    blockType: 'actor',
+    uniqueId: blockId,
+    parentId,
+    sourceId,
+    properties: {
+      src: sourceId,
+      id: actorId,
+    },
+  };
+}
+
+function getActorId(
+  ts: typeof import('typescript'),
+  parentId: string,
+  index: number,
+  idProperty: PropertyAssignment | undefined,
+) {
+  if (idProperty && ts.isStringLiteralLike(idProperty.initializer)) {
+    return idProperty.initializer.text;
+  }
+  // TODO: replace this with `:serializableId:invocation[:index]`
```

Comment:
why? even including `parentId` here is weird-ish to me. What happens when you reanchor transitions. The Studio does allow for this right now but I'm honestly not sure what the synced code should be after such a thing (and the exported code seems to ignore such transition - so I'd say this is a bug). 

Putting aside the reanchoring thing - the `serializableId` is would quickly become confusing and a lie here when you start reparenting states.

Reply:
`:serializableId:invocation[:index]` is the format the current extractor uses and Studio expects. As far as I can tell, no specific operation is done by the value of it so it can change.
Introducing parent id here is just temporary so that I had a way to ensure the block is extracted in the right place. It can be removed too.
I'm unsure what happens to the id after re-anchoring in Studio but I can check. For now, this can be any value.

Reply:
In a very similar manner even using `index` here is dubious. 

Simplified Input:
```ts
src: [a, b] // `b` gets assigned id with `[1]` in it
```

Studio actions:
1. remove `a`
2. add new invoke

Would those ever be updated? If not then we end up with a confusing thing because `b` is at the first position (index 0) but yet it uses `[1]` in its inline id

Reply:
https://github.com/statelyai/xstate-tools/pull/442/files#r1457510255